modules/virt-example-nmstate-IP-management.adoc

```diff
@@ -91,33 +91,156 @@ The following snippet configures an Ethernet interface that uses a dynamic IP ad
+By default, the `nmstate` API stores DNS values globally as against in a network interface. For certain situations, you must configure a network interface to store DNS values. To define a DNS configuration for a network interface, you must initially specify the `dns-resolver` section in the network interface's YAML configuration file as shown in the following example:
 
 [source,yaml]
 ----
 # ...
-    interfaces: <1>
-       ...
-       ipv4:
-         ...
-         auto-dns: false
-         ...
-    dns-resolver:
-      config:
-        search:
-        - example.com
-        - example.org
-        server:
-        - 8.8.8.8
+dns-resolver:
+  config:
+    search:
+      - example.com
+      - example.org
 # ...
 ----
-<1> You must configure an interface with `auto-dns: false` or you must use static IP configuration on an interface in order for Kubernetes NMState to store custom DNS settings.
+
+[TIP]
+====
+Setting a DNS configuration is comparable to modifying the `/etc/resolv.conf` file.
+====
+
+The following examples show situations that require configuring a network interface to store DNS values:
 
 [IMPORTANT]
 ====
-You cannot use `br-ex`, an OVNKubernetes-managed Open vSwitch bridge, as the interface when configuring DNS resolvers.
+You cannot use `br-ex` bridge, an OVNKubernetes-managed Open vSwitch bridge, as the interface when configuring DNS resolvers.
 ====
 
+* Configure a static DNS for a network interface with an automatic IP configuration. Note that for this configuration, you must set the `auto-dns` parameter to `false`, so that the Kubernetes NMState Operator can store custom DNS settings for the network interface.
++
+[source,yaml]
+----
+# ...
+    server:
+    - 2001:db8:f::1
+    - 192.0.2.251
+interfaces:
+  - name: eth1
+    type: ethernet
+    state: up
+    ipv4:
+      enabled: true
+      dhcp: true
+      auto-dns: false
+    ipv6:
+      enabled: true
+      dhcp: true
+      autoconf: true
+      auto-dns: false
+# ...
+----
+
+* Configure a static DNS for a network interface with an static IP configuration. Note that for this configuration, you must set the `auto-dns` parameter to `true`.
++
+[source,yaml]
+----
+# ...
+    server:
+    - 2001:4860:4860::8844
+    - 192.0.2.251
+interfaces:
+  - name: eth1
+    type: ethernet
+    state: up
+    ipv4:
+      enabled: true
+      dhcp: false
+      address:
+      - ip: 192.0.2.251
+        prefix-length: 24
+    ipv6:
+      enabled: true
+      dhcp: false
+      autoconf: false
+      address:
+      - ip: 2001:db8:1::1
+        prefix-length: 64
+routes:
+  config:
+  - destination: 0.0.0.0/0
+    next-hop-address: 192.0.2.1
+    next-hop-interface: eth1
+  - destination: ::/0
+    next-hop-address: 2001:db8:1::3
+    next-hop-interface: eth1
+# ...
+----
+
+* Configure a static DNS name server to append to a The Dynamic Host Configuration Protocol (DHCP) server:
```

Comment:
The IPv6 RA/autoconf can also provide DNS nameserver: https://datatracker.ietf.org/doc/html/rfc8106

Should be `DHCP and IPv6 RA/Autoconf or just DHCP/Autoconf`.

Reply:
Thanks. I'm unsure how to state "RA/Autoconf", as in:

* Router Advertisement (RA)/Autoconf server
* Router Advertisement (RA) or Autoconf-package-enabled servers

I cannot state a shorthand version until I spell out all acronyms.

Reply:
"append to Dynamic Host Configuration Protocol (DHCP) and IPv6 Router Advertisement (RA) servers", "servers" is wrong here?

Reply:
The [IPv6 RA(Router Advertisement)][0] is the action for router. The autoconf is form the view of network node (like linux system), defined in [RFC 4862 IPv6 Stateless Address Autoconfiguration][1] (aka, SLAAC). They are talking about the same thing, but from different view.

The correct way is state it as `IPv6 stateless autoconfiguration` or `IPv6 SLAAC`, but many administrator are more familiar with IPv6-RA.

[0]: https://datatracker.ietf.org/doc/html/rfc4861
[1]: https://datatracker.ietf.org/doc/html/rfc4862

Reply:
Thanks. Updated to state:

Configure a static DNS name server to append to Dynamic Host Configuration Protocol (DHCP) and IPv6 Stateless Address AutoConfiguration (SLAAC) servers: